### PR TITLE
Hollow Buttons

### PR DESCRIPTION
**Pull request by joeworkman**
Yeah! Hollow Buttons! Just add the `hollow` class

### CODE REVIEW

File: scss/components/_button.scss [hunk 1]
```diff
@@ -310,3 +310,47 @@ table.button.warning:hover {
...
+    &.rounded {
+      table a {border-radius:$global-rounded;}
+    }
+
+    @include hollowButton($primary-color);
+    @include hollowButton($secondary-color,'secondary');
+    @include hollowButton($alert-color,'alert');
+    @include hollowButton($warning-color,'warning');
+    @include hollowButton($success-color,'success');
+}
```
Comment (IamManchanda) [inline scss/components/_button.scss]: End of line!

Comment (joeworkman) [reply scss/components/_button.scss]: End of line?

Comment (IamManchanda) [reply scss/components/_button.scss]: Yup newline after the end of file please!

Comment (rafibomb) [reply scss/components/_button.scss]: @IamManchanda Seems like an older standard https://robots.thoughtbot.com/no-newline-at-end-of-file

What is the advantage to doing it that way?

File: scss/components/_button.scss [hunk 2]
```diff
@@ -310,3 +310,47 @@ table.button.warning:hover {
...
   }
 }
+
+// Hollow Buttons
+@mixin hollowButton($color,$class:false) {
+    // If this mixin is called with a class, then a class will be added
+    // Then the mixin calls itself with just the color to obtain the color styles
+    @if $class {
+        &.#{$class} {
+            @include ghostButton($color);
```
Comment (rafibomb) [inline scss/components/_button.scss]: should this be hollowButton?

Comment (joeworkman) [reply scss/components/_button.scss]: Damn! I renamed everything from Ghost to Hollow in order to follow the Sites framework. I will fix.

**Comment by joeworkman** [comment]
I pushed up the fixed include.

**Comment by rafibomb** [comment]
I pushed up an addition to the docs and added a test. Interestingly - there some clients it didn't work on at all - https://litmus.com/checklist/emails/public/836f9f6

We'll need to get them working in Outlook.com to get merged in. Is that something you can look at?

**Comment by joeworkman** [comment]
Sure. I will see if I can get it done next week.

**Comment by joeworkman** [comment]
So... I have discovered that if your link does not have an actual URL set as the href, Outlook.com will ignore certain link styles. Crazy!

I could duplicate the hollow button issues in Outlook.com when I had `href="#"`. However, once I set it to `href="https://emailstacks.com"` everything worked like gravy! 

With that said, I have found a couple small bugs in the hollow styles. I will push up changes today.

**Comment by joeworkman** [comment]
I have fixed the hollow buttons so that they are now working properly in all clients, including Outlook. Here is a link to my test... 

https://www.emailonacid.com/app/acidtest/display/summary/Z33aiBixtvJ2mHuAIP4NUma21RRVWWKpgCsFETzL9IdH0/shared

Some of my changes modify normal button styles in order to make sure that hollow buttons are sized the same as normal buttons. For example, tiny and small buttons had the padding added twice: once to the anchor and another to the td. This was causing issues with hollow buttons. Now everything is uniform.

File: scss/components/_button.scss [hunk 3]
```diff
@@ -303,10 +302,57 @@ table.button.warning:hover {
...
     }
 
     a {
       border: 0px solid darken($warning-color, 10%);
     }
   }
 }
+
+// Hollow Buttons
+@mixin hollowButton($color,$class:false) {
```
Comment (kball) [inline scss/components/_button.scss]: please use `kebab-case` for mixin names

Comment (joeworkman) [reply scss/components/_button.scss]: done

Comment (kball) [reply scss/components/_button.scss]: Also a little confused by the naming convention - it looks like this is not all of the hollow button styles, but simply the color-specific pieces... maybe we should have 2 mixins? `hollow-button-core` for what is now directly inline, and `hollow-button-color` for this mixin?

**Pull request merged**